servicex/servicex.py

```diff
@@ -149,39 +199,83 @@ async def get_data_async(selection_query: str, datasets: Union[str, List[str]],
                             them, and how to format them.
         datasets            Dataset or datasets to run the query against.
         service_endpoint    The URL where the instance of ServivceX we are querying lives
-        data_type           How should the data come back? 'pandas' and 'awkward' are the only
-                            legal values. Defaults to 'pandas'
+        data_type           How should the data come back? 'pandas', 'awkward', and 'root-file'
+                            are the only legal values. Defaults to 'pandas'
         image               ServiceX image that should run this.
         max_workers         Max number of workers that will run to process this request.
+        storage_directory   Location where files should be downloaded. If None is specified then
+                            they are stored in the machine's temp directory.
+        file_name_func      Returns a path where the file should be written. The path must be
+                            fully qualified (`storage_directory` must not be set if this is used).
+                            See notes on how to use this lambda.
+        redownload_files    If true, evne if the file is already in the requested location,
+                            re-download it.
 
     Returns:
-        df                  Pandas DataFrame that contains the resulting flat data, or an awkward
```

Comment:
Not a bad idea - no more magic strings. I put the string argument in there thinking the person calling might want to switch the output format, if you don't have a string (or other) argument, then you have to code a giant if statement. The return would definitely be nicer! Perhaps it could be by return type? files, awkward arrays, pandas arrays?

Regardless, I would move the changes off to refactoring, issue #22